sources/engine/Xenko.Graphics/Direct3D/CommandList.Direct3D.cs

```diff
@@ -255,6 +263,30 @@ internal void SetShaderResourceView(ShaderStage stage, int slot, GraphicsResourc
             shaderStages[(int)stage - 1].SetShaderResource(slot, shaderResourceView != null ? shaderResourceView.NativeShaderResourceView : null);
         }
 
+        /// <summary>
+        ///     Sets an unordered access view to the shader pipeline, without affecting ones that are already set.
+        /// </summary>
+        /// <param name="slot">The binding slot.</param>
+        /// <param name="shaderResourceView">The shader resource view.</param>
+        /// <param name="view">The native unordered access view.</param>
+        /// <param name="uavInitialOffset">The Append/Consume buffer offset. See SetUnorderedAccessView for more details.</param>
+        internal void OMSetSingleUnorderedAccssView(int slot, SharpDX.Direct3D11.UnorderedAccessView view, int uavInitialOffset)
+        {
+            currentUARenderTargetViews[slot] = view;
+
+            int remainingSlots = currentUARenderTargetViews.Length - currentRenderTargetViewsActiveCount;
+
+            var uavs = new SharpDX.Direct3D11.UnorderedAccessView[remainingSlots];
+            Array.Copy(currentUARenderTargetViews, currentRenderTargetViewsActiveCount, uavs, 0, remainingSlots);
+
+            var uavInitialCounts = new int[remainingSlots];
+            for (int i = 0; i < remainingSlots; i++)
+                uavInitialCounts[i] = -1;
+            uavInitialCounts[slot - currentRenderTargetViewsActiveCount] = uavInitialOffset;
+
+            outputMerger.SetUnorderedAccessViews(currentRenderTargetViewsActiveCount, uavs, uavInitialCounts);
```

Comment:
Note: double-checked into SharpDX code and it seems they use `D3D11_KEEP_RENDER_TARGETS_AND_DEPTH_STENCIL` which is what we want. All good!